pkg/controllers/user/rbac/globalrole_handler.go

```diff
@@ -70,9 +84,42 @@ func (c *grbHandler) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object
 			return obj, err
 		}
 	}
+
 	return obj, nil
 }
 
+// isAdminRole detects whether a GlobalRole has admin permissions or not.
+func (c *grbHandler) isAdminRole(rtName string) (bool, error) {
+	gr, err := c.grLister.Get("", rtName)
+	if err != nil {
+		return false, err
+	}
+
+	// global role is admin
+	if gr.Name == "admin" {
```

Comment:
can we also add a check for "Builtin" like we do for cluster-owner?

Reply:
Ahh yes, good point, let me add that